Userland/Libraries/LibWeb/WebDriver/Screenshot.cpp

```diff
@@ -61,6 +61,10 @@ Response capture_element_screenshot(Painter const& painter, Page& page, DOM::Ele
         auto canvas_element = DOM::create_element(element.document(), HTML::TagNames::canvas, Namespace::HTML).release_value_but_fixme_should_propagate_errors();
         auto& canvas = verify_cast<HTML::HTMLCanvasElement>(*canvas_element);
 
+        // FIXME: Handle DevicePixelRatio in HiDPI mode.
+        MUST(canvas.set_width(rect.width()));
+        MUST(canvas.set_height(rect.height()));
```

Comment:
does that mean that somehow we do not always init size of `HTMLCanvasElement` to default values?

Reply:
We do init it to the default values, which is the problem. When we call create bitmap below, those are actually "minimum required" dimensions .The code will actually end up creating a bitmap that is always at least 150 pixels in height, and at least 300 pixels in width (per your correction to the default sizes, I'll push that commit description error fix soon :) ). In the case of WPT's infrastructure/assumptions/canvas-background.html test, the document element is only 48 to 50 pixels tall. So the "test" image will have the top 48 pixels of the background painted only, but create a data URL for a 150 pixel tall image. As for the reference image, it gets filled in with white for some reason 🤷‍♂️. I'm not sure what the infra is doing behind the scenes. But, setting the size of the canvas to the size of the rect intersected with the viewport makes the test (and the float tests) pass.